src/lib/auth/auth.ts

```diff
@@ -0,0 +1,127 @@
+/* Facade for keycloak */
+import Keycloak, { KeycloakInitOptions, KeycloakInstance } from 'keycloak-js';
+
+const toPromise = keycloakPromise =>
+  new Promise<any>((resolve, reject) => keycloakPromise.success(resolve).error(reject));
+
+export interface ResourceRole {
+  resource: string;
+  resourceId: string;
+  role: string;
+}
+
+export interface AuthServiceConfiguration {
+  oidcIssuer: string;
+  clientId: string;
+  redirectUri: string;
+  logoutRedirectUri: string;
+}
+
+export interface User {
+  username: string;
+  name: string;
+}
+
+const extractUser: (string) => User = tokenParsed => ({ username: tokenParsed.user_name, name: tokenParsed.name });
+
+const extractResourceRoles: (authorities: string) => ResourceRole[] = (authorities: string) => {
```

Comment:
Should be a private function of auth class

Reply:
det er egentlig static, men til å peke static funksjon er lengere, så jeg valgt å ikke gjøre det